docs(ai): clarify upstream branch policy for CI reliability (#2806)
eic/EICrecon: AGENTS.md

## Working Effectively with EICrecon

### PR Branch Hosting Policy

> Prefer pushing feature branches to the upstream `eic/EICrecon` remote and opening PRs from those upstream-hosted branches. Avoid fork-based PR branches by default, because CI execution reliability depends on upstream-hosted branches.

Reviewer: ```suggestion
If pull request needs to be opened, feature branches is to be pushed to the upstream `eic/EICrecon` and not to a fork.
```
Let's not encourage AI to push branches without asking.

Author: ```suggestion
If pull requests need to be opened, prefer pushing feature branches to the upstream `eic/EICrecon` remote and opening PRs from those upstream-hosted branches. Avoid fork-based PR branches by default, because CI execution reliability depends on upstream-hosted branches.
```

Author: Not sure this is really going to stop anyone's AI from doing what it wants to do...
